File: src/types/product.ts
export type ComboOption = {
    id: string;
    name: string;
    price: number;
    image?: string;
    category: 'accessory' | 'quantity';
}

export type FlowerQuantity = {
    stems: number;
    priceModifier: number;
}

export type Product = {
    id: string | number;
    name: string;
    description: string;
    basePrice: number;
    images: string[];
    category: 'roses' | 'peonies' | 'daisies' | 'mixed' | 'wildflowers';
    ecoFriendly: boolean;
    sustainabilityInfo?: string;
    comboOptions: ComboOption[];
    quantityOptions: FlowerQuantity[];
    inStock: boolean;
}

export type CartItem = {
    product: Product;
    quantity: number;
    selectedCombos: ComboOption[];
    selectedQuantity: FlowerQuantity | null;
    totalPrice: number;
}

export type CartState = {
    items: CartItem[];
    addItem: (item: Omit<CartItem, 'totalPrice'>) => void;
    removeItem: (productId: string | number) => void;
    updateQuantity: (productId: string | number, quantity: number) => void;
    clearCart: () => void;
    getTotal: () => number;
    getItemCount: () => number;
}

File: backend/FlowHerAPI/Models/Product.cs
namespace FlowHerAPI.Models;

public class Product
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public decimal BasePrice { get; set; }
    public string Category { get; set; } = string.Empty;
    public bool EcoFriendly { get; set; }
    public string? SustainabilityInfo { get; set; }
    public bool InStock { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<ProductImage> Images { get; set; } = new List<ProductImage>();
    public ICollection<ComboOption> ComboOptions { get; set; } = new List<ComboOption>();
    public ICollection<QuantityOption> QuantityOptions { get; set; } = new List<QuantityOption>();
}
